### PR TITLE
feat(web-anaytics): Add extra redis functions (SCard, SAddNX, buffers)

### PR DESCRIPTION
**Pull request by robbie-c**
## Problem

Split out of https://github.com/PostHog/posthog/pull/25915/files to make it easier to review

## Changes
Add a few extra redis functions
* SCARD
* SADD + SCARD in a transaction
* getting and setting buffers
* set NX (only set if a value is not set already)

## Does this work well for both Cloud and self-hosted?

Yes	

## How did you test this code?

Added some tests

### CODE REVIEW

File: plugin-server/src/utils/db/db.ts [hunk 1]
```diff
@@ -254,6 +270,49 @@ export class DB {
...
+        value: unknown,
+        tag: string,
+        ttlSeconds?: number,
+        options: CacheOptions = {}
+    ): Promise<'OK' | null> {
+        const { jsonSerialize = true } = options
+
+        return instrumentQuery('query.redisSetNX', tag, async () => {
+            const client = await this.redisPool.acquire()
+            const timeout = timeoutGuard('Setting redis key delayed. Waiting over 30 sec to set key (NK)', { key })
```
Comment (rafaeelaudibert) [inline plugin-server/src/utils/db/db.ts:301]:
```suggestion
            const timeout = timeoutGuard('Setting redis key delayed. Waiting over 30 sec to set key (NX)', { key })
```

**Review by rafaeelaudibert — APPROVED** [review]
LGTM, just a minor typo in one of the logs :)

File: plugin-server/src/utils/db/db.ts (hunk 1 above)
Comment (robbie-c) [reply plugin-server/src/utils/db/db.ts]: Ouch, good spot

**Review by mariusandra — APPROVED** [review]
LGTM

**Pull request merged**